### ISSUE TITLE
Reflectometry GUI fixes

### ISSUE THREAD

**Issue by gemmaguest** [mantidproject/mantid#21167]
This is an umbrella issue to keep track of bug fixes and minor features required for the new reflectometry GUI

### Transmission runs
We have had continual problems with providing transmission runs via the Settings tab. The bugs need to be fixed as a priority and we should add more unit tests to make sure we don't break this again in the future. It is also a priority to have some way of specifying transmission runs on a per-angle basis.

We may want to look at this in conjunction with #21218, which does some refactoring to make working with this code much easier.

- [x] (fixed by #21417) Transmission runs entered on the Settings tab are not found #21076
- [x] (fixed by #21417) You should be able to specify the transmission run on the settings tab or runs tab as a run number or a workspace name in the ADS - currently this works on the Runs tab but gives an unhandled exception on the settings tab.
- [x] (fixed by #21495) No parameters are getting passed through to CreateTransmissionWorkspaceAuto #21457
- [x] (fixed by #21762) @gemmaguest Allow transmission runs to be specified on a per-angle basis #20088 

### Other
- [x] (fixed by #21808) Improve time sliced workspace names #21004
- [x] Output workspace names should have suffix instead of prefix #21004
- [x] ~~(#21317 helps with this)~~ When adding a new group to the table, the new group has a single child row, which is where users usually want to start entering run numbers. However, the new row is hard to see and the group is collapsed by default, which users find confusing. We should do something to help prompt users where to enter text, e.g. expand the group, select the child row and focus the first cell ready for editing.
- [x] (fixed by #21417) Properties on the Runs tab should take precedence over properties on the Settings tab

## Priority - Quick wins
These changes are considered important and should be relatively easy:

### General
- [x] (fixed by #21680) Editing settings on the Events and Settings tabs should be disabled during processing to improve robustness and avoid confusion.

### Parameters on the GUI
- [x] (fixed by #21364) Expose the Q-summation functionality on the Settings tab (via `SummationType` and `ReductionType` drop downs as on the RROA dialog). #21353 
- [x] (fixed by #21343) Expose the `CorrectDetectors` tick box on the Settings tab similarly to the RROA dialog. #21337
- [x] (fixed by #21314) Remove the DirectBeam box from the GUI and the underlying `RegionOfDirectBeam` property from `ReflectometryReductionOneAuto`. Scientists at ISIS do not use this and don't think it would work well anyway. #21313
- [x] (fixed by #21622) The `Get Defaults` button on the Settings tab doesn't load some properties from the parameter file, e.g. `DetectorCorrectionType`. I think there may be something we need to enable in the code for this. Scientists would like to be able to set all of the algorithms' properties from the parameter file. #21543

### Help
- [ ] @edwardb96 ~~(partially fixed by #21668; save tab waiting on #21716)~~ We should make sure all elements have tooltips or What's This help available where possible (preferably the former). 
- [x] @edwardb96 (fixed by #21668) We should have a help button link to the wiki.

## New features
These are more significant changes which may not get done for v3.12:

- [x]  Polarisation correction properties should be taken from the params file #22500
- [x] @edwardb96 (fixed by #21716) The output data should be automatically saved to ASCII. You should be prompted for the save directory if it is not set on the ASCII tab or if the path is not writable. There should be a tickbox to disable auto-saving (primarily for use when testing where the prompt may become annoying). Base default path for the prompt on the general save directory from the user directories dialog? Note there is currently a bug on the save tab where, if you enter an incorrect path, mantid catches an unexpected exception (file not found) when you hit Save. #21691
- [x] Allow transmission runs to be summed before combining #21530. ~~We should finish #20088 first.~~

## Other issues
These were spotted during developer testing. They are not considered a priority because they haven't been highlighted as important by users/scientists. However, some of them may need to be addressed as part of other work.

### General
- [x] (fixed by PR #22443) Pausing reduction will immediately make the `process` button immediately available (before reduction itself has actually paused). If `process` is clicked before reduction has finished, a critical error would be thrown. `Process` should be re-enabled only when reduction has finished, not just when the pause button is clicked.
- [x] Now that the Settings tab takes priority, the Stitch1DMany Params are ignored because they are overridden by `dQ/Q` from the Runs tab.
- [x] If slit resolution cannot be calculated and `dQ/Q` is not provided then you get error messages in the log (as expected) but an unhelpful pop up messages `Invalid value for property InputWorkspaces (str list) "": Select a value`
- [ ] #21224 GUI crashes when overwriting table
- [x] @gemmaguest (fixed by #21762) When you transfer new runs into the table, any existing runs that had already been reduced will lose their green highlighting and will be unnecessarily re-processed.
- [ ] There is a minor bug where if you click Get Defaults on the Settings tab, and then edit the parameters file to remove a default that previously existed, when you subsequently hit Get Defaults it re-supplies the previous default even though it no longer exists.
- [ ] The behaviour of the plot buttons is unintuitive #22034 
- [x] If you select a sub-set of the rows in a group and process them, then that subset is post-processed. Should this be allowed? It seems unintuitive as the purpose of a group is to do the post-processing, and this may cause maintenance problems going forward.
- [x] (fixed in PR #22443) If the processing algorithm fails (e.g. if slits cannot be found in the IDF), then the row is marked as processed. If it fails because the run is not found, then it is not marked as processed. It would be better in both cases to have some indication that processing has been attempted and has failed.
- [ ] No prompt to save changes #22051
- [x] Pasting from excel doesn't work correctly #22050

### Event handling
- [x] @gemmaguest (fixed as part of #20088) Plotting rows in event handling mode plots the unbinned workspace, rather than the binned one (as for non-event mode)
- [x] (fixed by #22443) Highlighting of processed rows is not done in event handling mode.
- [x] Event slicing doesn't make sense for multiple input runs, so if you provide e.g. INTER 13461+13462 and set, say, uniform slicing = 2 then it fails. This is what we expect, but you get an unhelpful "some errors were encountered" error because [this](https://github.com/mantidproject/mantid/blob/master/qt/scientific_interfaces/ISISReflectometry/ReflDataProcessorPresenter.cpp#L164) throws. This is because runs are pre-loaded by code that assumes the input is a single run number. You can also get failures due to leading whitespace not being trimmed.
- [x] Async operation is not done in event handling mode, so processing a large number of runs locks up the whole mantidplot gui until finished. I think we should hold off on this for now because the async operation needs work anyway.
- [x] Stitching more than 2 time-sliced workspaces with different numbers of slices fails, e.g. process INTER runs 13460, 13461 and 13462 with a slice interval of 500 seconds. You get 3, 2 and 5 slices respectively and processing the group gives an error when stitching.
- [x] If you process a non-event workspace/group with event handling enabled, then it is processed ok as non-event data but plotting the results doesn't work. I think the code is assuming slices exist and looking for the child slices. If you run a successful sliced reduction and then change the run number to a non event workspace, the slices list is not cleared, so plotting plots the previous run's sliced workspaces!

### GenerateNotebook
- [x] (fixed by #21495) Code generated for a notebook doesn't work due to unquoted values being passed to `ReflectometryReductionOneAuto`
- [x] (fixed by #21495) There is a bug in setting the legend location in a generated notebook which includes post-processing.
- [x] (fixed by #21495) `GenerateNotebook` doesn't support special characters such as `+` in workspace names
- [ ] Inconsistencies between the stitching params used in MantidPlot and the output notebook #21501
- [ ] Changes on the settings tab are not picked up when regenerating a notebook unless changes were also made on the data processor table. I think this is related to the fact that the flag indicating they have been processed is not reset when changing the settings (there is a related issue where rows should have green highlighting removed when settings are changed to indicate that they will be re-processed). 
- [ ] Generating a notebook does not produce sensible results if you use names of input workspaces that are already loaded in the ADS - the code always assumes the input is a run number that needs to be loaded. It's not straightforward to extract the run number(s) because the workspace may have been preprocessed. The best way around this is probably to extract the history for the input workspaces if they already exist in the ADS and use that to recreate the workspace in the notebook.
- [ ] The wrong load algorithm is used by `GenerateNotebook` - it is hard coded to load runs using `Load`, when actually a different algorithm (e.g. `LoadISISNexus`) can be specified in the data processor.
- [ ] The output from `GenerateNotebook` gives an error when trying to plot the stitched workspace when using workspace groups, e.g. for polref 14885 and 14886

**Comment by gemmaguest** [comment]
@edwardb96 We'll need to coordinate who does what on this list. I'm going to start by looking at the transmission runs stuff.

**Comment by edwardb96** [comment]
Is "a recent regression where no parameters from the GUI are getting passed through to CreateTransmissionWorkspaceAuto" fixed by https://github.com/mantidproject/mantid/pull/21237?

**Comment by gemmaguest** [comment]
Yes, I think so: #21237 fixes an error about "FirstTransmissionRun not found" and the parameters seem to be getting passed through ok to CreateTransmissionWorkspaceAuto.

**Comment by gemmaguest** [comment]
Changed to med priority because the high priority fixes are now complete.

**Comment by gemmaguest** [comment]
Cleared milestone because most fixes in here are not high priority

**Comment by gemmaguest** [comment]
The slicing and async issues are resolved as part of the restructuring in #23051. The stitching error with different numbers of slices is probably expected.

**Comment by gemmaguest** [comment]
Closing because the remaining issues are no longer relevant on the restructured GUI
